packages/mobile/src/home/WalletHome.tsx

```diff
@@ -76,13 +89,22 @@ interface State {
 export class WalletHome extends React.Component<Props, State> {
   scrollPosition: Animated.Value<number>
   onScroll: (event: NativeSyntheticEvent<NativeScrollEvent>) => void
+  isAccountBalanceZero: boolean
 
   constructor(props: Props) {
     super(props)
 
     this.scrollPosition = new Animated.Value(0)
     this.onScroll = Animated.event([{ nativeEvent: { contentOffset: { y: this.scrollPosition } } }])
-    this.state = { isMigrating: false }
+    this.state = {
+      isMigrating: false,
+    }
+
+    const hasStable = STABLE_CURRENCIES.some((currency) =>
+      props.balances[currency]?.isGreaterThan(STABLE_TRANSACTION_MIN_AMOUNT)
+    )
+    const hasGold = props.balances[Currency.Celo]?.isGreaterThan(GOLD_TRANSACTION_MIN_AMOUNT)
+    this.isAccountBalanceZero = !hasStable && !hasGold
```

Comment:
Why did you decide to go with this approach instead of the one suggested in [this previous comment](https://github.com/valora-inc/wallet/pull/1212#discussion_r719911404)?

Reply:
I see. Then can we move this logic into `shouldShowCashInBottomSheet` and get rid of the global `isAccountBalanceZero`?